src/server/server.cpp

```diff
@@ -480,6 +478,22 @@ auto mir::Server::x11_display() const -> mir::optional_value<std::string>
     BOOST_THROW_EXCEPTION(std::logic_error("Cannot open connection when not running"));
 }
 
+auto mir::Server::get_activation_token() const -> std::optional<std::string>
+{
+    if (auto const config = self->server_config)
+    {
+        auto const& connector = std::dynamic_pointer_cast<mir::frontend::WaylandConnector>(config->the_wayland_connector());
+        auto xdg_activation_impl = std::static_pointer_cast<mir::frontend::XdgActivationV1>(connector->get_extension("xdg_activation_v1"));
```

Comment:
We shouldn't need to break abstractions like this. (When you find yourself writing code like this, stop and reconsider the approach)